recipes/pdcurses/all/conanfile.py

```diff
@@ -1,149 +1,197 @@
-    topics = ("conan", "pdcurses", "curses", "ncurses")
+    license = "LicenseRef-LICENSE"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://pdcurses.org/"
-    license = "Unlicense", "MITX", "CC-BY-4.0", "GPL", "FSFUL"
+    topics = ("curses", "ncurses")
 
+    package_type = "library"
     settings = "os", "arch", "compiler", "build_type"
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
         "enable_widec": [True, False],
-        "with_sdl": [None, "1", "2"],
+        "with_sdl": [True, False],
+        "with_x11": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
         "enable_widec": False,
-        "with_sdl": None,
+        "with_sdl": False,
+        "with_x11": True,
     }
 
-    _autotools = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
-        if self.settings.os not in ("FreeBSD", "Linux"):
-            del self.options.enable_widec
+        if self.settings.os not in ["FreeBSD", "Linux"]:
+            del self.options.with_x11
+        if is_apple_os(self):
+            # Only the sdl2 subsystem is supported on macOS
+            self.options.with_sdl = True
 
     def configure(self):
-        if tools.is_apple_os(self.settings.os):
-            raise ConanInvalidConfiguration("pdcurses does not support Apple")
-        if self.options.with_sdl:
-            raise ConanInvalidConfiguration("conan-center-index has no packages for sdl (yet)")
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.cppstd
-        del self.settings.compiler.libcxx
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def requirements(self):
-        if self.settings.os in ("FreeBSD", "Linux"):
+        if self.options.with_sdl:
+            self.requires("sdl/2.28.3", transitive_libs=True)
+        if self.options.get_safe("with_x11"):
             self.requires("xorg/system")
 
+    def validate(self):
+        if self.options.with_sdl and self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("with_sdl option is not yet supported on Windows")
+        if self.settings.os != "Windows" and not self.options.get_safe("with_x11") and not self.options.with_sdl:
+            raise ConanInvalidConfiguration("At least one of with_x11 or with_sdl options must be enabled")
+        if cross_building(self):
+            raise ConanInvalidConfiguration("Cross-building is not supported")
```

Comment:
How cross_building is not working only now? Please, add more details.

Reply:
I don't think the cross-building was ever tested. The most recent PR did not building for apple-clang at all: https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/5300/2/pdcurses/3.9//summary.json

It's a Makefile-only project and does not have the standard cross-compiling functionality of autotools, but I suppose it could be adapted for cross-compilation with some work. Better left for a separate PR, though.